state/core_access.go

```diff
@@ -455,6 +521,48 @@ func (ca *CoreAccessor) QueryRedelegations(
 	})
 }
 
+func (ca *CoreAccessor) getMinGasPrice(ctx context.Context) (float64, error) {
+	ca.mtx.Lock()
+	defer ca.mtx.Unlock()
+	if ca.minGasPrice < 0 {
+		minGasPrice, err := ca.queryMinimumGasPrice(ctx)
+		if err != nil {
+			return -1, fmt.Errorf("querying minimum gas price: %w", err)
+		}
+		ca.minGasPrice = minGasPrice
+	}
+	return ca.minGasPrice, nil
+}
+
+func (ca *CoreAccessor) setMinGasPrice(minGasPrice float64) {
+	ca.mtx.Lock()
+	defer ca.mtx.Unlock()
+	ca.minGasPrice = minGasPrice
+}
+
+func (ca *CoreAccessor) markSuccessfulPFB() {
+	ca.mtx.Lock()
+	defer ca.mtx.Unlock()
+	ca.lastPayForBlob = time.Now().UnixMilli()
+	ca.payForBlobCount++
+}
+
+// QueryMinimumGasPrice returns the minimum gas price required by the node.
+func (ca *CoreAccessor) queryMinimumGasPrice(
+	ctx context.Context,
+) (float64, error) {
+	rsp, err := nodeservice.NewServiceClient(ca.coreConn).Config(ctx, &nodeservice.ConfigRequest{})
+	if err != nil {
+		return 0, err
+	}
```

Comment:
I didn't look into the impl of it, but what if we construct it only once on start?

Reply:
We can if you want but I assumed this was pretty lightweight (i.e. just boilerplate code doing the encoding/decoding)

Reply:
Ok then